### PR TITLE
Updated eventing to have new filters in beta

### PR DESCRIPTION
**Pull request by Cali0707**
Part of #3363 

## Proposed Changes

- Update the eventing dependency to the commit where we moved the new filters to enabled by default and beta
-
-

**Release Note**

```release-note
New trigger filters are now beta and enabled by default.
```

**Docs**

### CODE REVIEW

**Comment by pierDipi** [comment]
@Cali0707 you need to re-run `hack/update-codegen.sh`

**Comment by Cali0707** [comment]
/cc @pierDipi 
I updated the codegen

**Comment by Cali0707** [comment]
@pierDipi @creydr I'm not 100% sure but I think that the tests may be failing due to some form of OIDC issue with the triggers. See the snippet from the artifact dump:
```
>>> Listing
NAMESPACE   NAME      BROKER   SUBSCRIBER_URI                           AGE     READY     REASON
sacura      trigger   broker   http://sacura.sacura.svc.cluster.local   3m34s   Unknown   
>>> Details
apiVersion: v1
items:
- apiVersion: eventing.knative.dev/v1
  kind: Trigger
  metadata:
    annotations:
      eventing.knative.dev/creator: prow-job@knative-tests.iam.gserviceaccount.com
      eventing.knative.dev/lastModifier: prow-job@knative-tests.iam.gserviceaccount.com
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"eventing.knative.dev/v1","kind":"Trigger","metadata":{"annotations":{},"name":"trigger","namespace":"sacura"},"spec":{"broker":"broker","subscriber":{"ref":{"apiVersion":"v1","kind":"Service","name":"sacura","namespace":"sacura"}}}}
    creationTimestamp: "2023-10-02T17:50:24Z"
    finalizers:
    - kafka.triggers.eventing.knative.dev
    generation: 1
    labels:
      eventing.knative.dev/broker: broker
    name: trigger
    namespace: sacura
    resourceVersion: "12821"
    uid: e8302bb4-2090-42e4-b233-766e58f83bbb
  spec:
    broker: broker
    filter: {}
    subscriber:
      ref:
        apiVersion: v1
        kind: Service
        name: sacura
        namespace: sacura
  status:
    annotations:
      group.id: knative-trigger-sacura-trigger
    conditions:
    - lastTransitionTime: "2023-10-02T17:50:26Z"
      status: "True"
      type: BrokerReady
    - lastTransitionTime: "2023-10-02T17:50:26Z"
      message: No dead letter sink is configured.
      reason: DeadLetterSinkNotConfigured
      status: "True"
      type: DeadLetterSinkResolved
    - lastTransitionTime: "2023-10-02T17:50:26Z"
      status: "True"
      type: DependencyReady
    - lastTransitionTime: "2023-10-02T17:50:24Z"
      status: Unknown
      type: OIDCIdentityCreated
    - lastTransitionTime: "2023-10-02T17:50:26Z"
      status: Unknown
      type: Ready
    - lastTransitionTime: "2023-10-02T17:50:26Z"
      message: 'Subscriber will receive events with the delivery order: UNORDERED'
      reason: SubscriberResolved
      status: "True"
      type: SubscriberResolved
    - lastTransitionTime: "2023-10-02T17:50:26Z"
      status: "True"
      type: SubscriptionReady
    observedGeneration: 1
    subscriberUri: http://sacura.sacura.svc.cluster.local
kind: List
metadata:
  resourceVersion: ""
```
Any ideas how to fix this?

File: control-plane/pkg/reconciler/trigger/trigger.go [hunk 1]
```diff
@@ -144,6 +144,13 @@ func (r *Reconciler) reconcileKind(ctx context.Context, trigger *eventing.Trigge
 		return nil
 	}
 
+	if feature.FromContext(ctx).IsOIDCAuthentication() {
+		return fmt.Errorf("OIDC is not implemented in the eventing kafka broker yet")
+	} else {
+		trigger.Status.Auth = nil
+		trigger.Status.MarkOIDCIdentityCreatedSucceededWithReason(fmt.Sprintf("%s feature disabled", feature.OIDCAuthentication), "")
+	}
+
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/trigger/trigger.go:153]: I think we want to use https://github.com/knative/eventing/pull/7316 for this

File: control-plane/pkg/reconciler/trigger/namespaced_trigger_test.go [hunk 2]
```diff
@@ -129,6 +129,7 @@ func namespacedTriggerReconciliation(t *testing.T, format string, env config.Env
 						withTriggerSubscriberResolvedSucceeded(contract.DeliveryOrder_UNORDERED),
 						withTriggerStatusGroupIdAnnotation(TriggerUUID),
 						reconcilertesting.WithTriggerDeadLetterSinkNotConfigured(),
+						reconcilertesting.WithTriggerOIDCIdentityCreatedSucceededBecauseOIDCFeatureDisabled(),
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/trigger/namespaced_trigger_test.go:132]: I just realized how long this function name is

File: control-plane/pkg/reconciler/trigger/namespaced_trigger_test.go [hunk 3]
```diff
@@ -129,7 +129,7 @@ func namespacedTriggerReconciliation(t *testing.T, format string, env config.Env
 						withTriggerSubscriberResolvedSucceeded(contract.DeliveryOrder_UNORDERED),
 						withTriggerStatusGroupIdAnnotation(TriggerUUID),
 						reconcilertesting.WithTriggerDeadLetterSinkNotConfigured(),
-						reconcilertesting.WithTriggerOIDCIdentityCreatedSucceededBecauseOIDCFeatureDisabled(),
+						reconcilertesting.WithTriggerOIDCIdentityCreatedNotSupported(),
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/trigger/namespaced_trigger_test.go]: We don't need this call, it's done in `reconcilertesting.WithInitTriggerConditions`

File: control-plane/pkg/reconciler/trigger/trigger.go [hunk 4]
```diff
@@ -87,6 +87,7 @@ type Reconciler struct {
 }
 
 func (r *Reconciler) ReconcileKind(ctx context.Context, trigger *eventing.Trigger) reconciler.Event {
+	trigger.Status.MarkOIDCIdentityCreatedNotSupported()
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/trigger/trigger.go:90]: Do we have an issue tracking this?

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

File: control-plane/pkg/reconciler/trigger/trigger.go (hunk 4 above)
Comment (Cali0707) [reply control-plane/pkg/reconciler/trigger/trigger.go:90]: I just made https://github.com/knative-extensions/eventing-kafka-broker/issues/3378

**Pull request merged**